shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-nacos/src/main/java/org/apache/shardingsphere/mode/repository/cluster/nacos/listener/NamingEventListener.java

```diff
@@ -0,0 +1,129 @@
+            List<Instance> instances = namingEvent.getInstances().stream().sorted(Comparator.comparing(MetadataUtil::getKey)).collect(Collectors.toList());
+            List<WatchData> watchDataList = new LinkedList<>();
+            instances.forEach(instance -> prefixListenerMap.forEach((prefixPath, listener) -> {
+                String key = MetadataUtil.getKey(instance);
+                if (key.startsWith(prefixPath)) {
+                    Instance preInstance = preInstances.remove(key);
+                    WatchData watchData = new WatchData(key, preInstance, instance, listener);
+                    watchDataList.add(watchData);
+                }
+            }));
+            preInstances.values().stream().sorted(Comparator.comparing(MetadataUtil::getKey).reversed()).forEach(instance -> prefixListenerMap.forEach((prefixPath, listener) -> {
+                String key = MetadataUtil.getKey(instance);
+                if (key.startsWith(prefixPath)) {
+                    WatchData watchData = new WatchData(key, instance, null, listener);
+                    watchDataList.add(watchData);
+                }
+            }));
+            watchDataList.forEach(watchData -> {
+                String key = watchData.getKey();
+                Instance preInstance = watchData.getPreInstance();
+                Instance instance = watchData.getInstance();
+                DataChangedEventListener listener = watchData.getListener();
+                DataChangedEvent.Type changedType = getEventChangedType(preInstance, instance);
+                switch (changedType) {
+                    case ADDED:
+                    case UPDATED:
+                        listener.onChange(new DataChangedEvent(key, MetadataUtil.getValue(instance), changedType));
+                        break;
+                    case DELETED:
+                        listener.onChange(new DataChangedEvent(key, MetadataUtil.getValue(preInstance), changedType));
+                        break;
+                    default:
+                }
+            });
+            setPreInstances(instances);
+        }
+    }
+    
+    private DataChangedEvent.Type getEventChangedType(final Instance preInstance, final Instance instance) {
+        DataChangedEvent.Type type;
+        if (Objects.isNull(preInstance) && Objects.nonNull(instance)) {
+            type = DataChangedEvent.Type.ADDED;
+        } else if (Objects.nonNull(preInstance) && Objects.nonNull(instance)
+                && MetadataUtil.getTimestamp(preInstance) != MetadataUtil.getTimestamp(instance)) {
+            type = DataChangedEvent.Type.UPDATED;
+        } else if (Objects.nonNull(preInstance) && Objects.isNull(instance)) {
+            type = DataChangedEvent.Type.DELETED;
+        } else {
+            return DataChangedEvent.Type.IGNORED;
+        }
+        return type;
+    }
+    
+    /**
+     * Update preInstances.
+     * @param instances instances
```

Comment:
Please keep blank line between java doc and @param

Reply:
OK.

Reply:
> Please keep blank line between java doc and @param

OK